pre_commit/languages/swift.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import unicode_literals
+
+import os
+
+from pre_commit.languages import helpers
+from pre_commit.util import clean_path_on_failure
+from pre_commit.xargs import xargs
+
+ENVIRONMENT_DIR = 'swift_env'
+BUILD_DIR = '.build'
+BUILD_CONFIG = 'release'
+
+
+def install_environment(
+        repo_cmd_runner,
+        version='default',
+        additional_dependencies=(),
+):
+    directory = repo_cmd_runner.path(helpers.environment_dir(
+        ENVIRONMENT_DIR, 'default',
+    ))
+
+    # Build the swift package
+    with clean_path_on_failure(directory):
+        os.mkdir(directory)
+        repo_cmd_runner.run((
+            'swift', 'build',
+            '-C', '{prefix}',
+            '-c', BUILD_CONFIG,
+            '--build-path', os.path.join(directory, BUILD_DIR),
+        ))
+
+
+def run_hook(repo_cmd_runner, hook, file_args):
+    directory = repo_cmd_runner.path(helpers.environment_dir(
+        ENVIRONMENT_DIR, 'default',
+    ))
+    cmd = helpers.to_cmd(hook)
+    full_binary_path = os.path.join(directory, BUILD_DIR, BUILD_CONFIG, cmd[0])
```

Comment:
instead of doing this, let's do a `PATH` envcontext (see any of the python / ruby / js ones for an example).  Then you can just use `helpers.to_cmd` directly